src/System.Memory/tests/ReadOnlyBuffer/ReadOnlyBufferFactory.cs

```diff
@@ -0,0 +1,133 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Collections.Generic;
+using System.Text;
+
+namespace System.Memory.Tests
+{
+    public abstract class ReadOnlyBufferFactory
+    {
+        public static ReadOnlyBufferFactory Array { get; } = new ArrayTestBufferFactory();
```

Comment:
I'd prefer that these subclasses not conflict with known system types like "Array" (and thus not require full qualification to get access to the real `Array` type.)